src/common/legacy/src/convert_function_to_cnn_network.cpp

```diff
@@ -401,10 +401,12 @@ void CNNLayerCreator::on_adapter(const std::string& name,
         params[name] = details::convertPrecision(type).name();
     } else if (auto a = ::ngraph::as_type<::ngraph::AttributeAdapter<::ngraph::PartialShape>>(&adapter)) {
         std::string dims;
-        auto shape = static_cast<::ngraph::PartialShape&>(*a);
-        for (int64_t i = 0; i < shape.rank().get_length(); i++) {
-            if (!dims.empty()) dims += ",";
-            dims += std::to_string(shape[i].get_length());
+        auto shape = a->get();
+        if (!shape.rank().is_dynamic()) {
```

Comment:
I think we need to throw an exception in case if shape is dynamic.
Also this check for dynamic rank looks incomplete, because later we do not check dimension for dynamism.